ferrocene/doc/evaluation-report/src/rustfmt/use-cases.rst

```diff
@@ -6,239 +6,70 @@
+**Output:** The rustfmt binary in the Ferrocene directory.
 
 **Environment constraints:** Tar is correctly installed.
 
 **Description:**
 
-1. The `user` downloads all the archives needed to install Ferrocene.
+1. The `user` downloads the rustfmt archive.
 
-2. The `user` extracts each archive into the installation directory using `tar`::
+2. The `user` extracts the archive into the installation directory using `tar`::
 
     tar -C path/to/installation -xf path/to/archive.tar.xz
 
-Building a Library
-------------------
-
-.. id:: RUSTC_UC1_RLIB
-
-**Actor(s):** User, rustc.
-
-**Input:** A Rust compilation unit.
-
-**Output:** A static Rust library.
-
-**Environment constraints:** Ferrocene is correctly installed and the
-environment is correctly set.
-
-**Description:**
-
-1. The `user` calls `rustc` with the following command line arguments::
-
-    --edition 2021
-    --crate-type rlib
-    <path>
-    # where <path> is the path to the root of the compilation unit, as a
-    # positional argument.
-
-2. `rustc` parses the command line arguments.
-
-3. `rustc` parses the Rust compilation unit.
-
-4. `rustc` analyzes the Rust compilation unit.
-
-5. `rustc` generates LLVM IR for the Rust compilation unit.
-
-6. `rustc` invokes `LLVM`, passing the generated LLVM IR along with
-   LLVM-related arguments.
-
-7. `LLVM` generates a static Rust library.
-
+Format a file
+-------------
 
-.. id:: RUSTC_UC2_STATICLIB
+.. id:: RUSTFMT_UC1_FMT
 
-**Actor:** User, rustc.
+**Actor(s):** User, rustfmt.
 
-**Input:** A Rust compilation unit.
+**Input:** One or multiple Rust source files.
 
-**Output:** A C-compatible static library.
+**Output:** Rust source files with the same paths, potentially modified.
 
-**Environment constraints:** Ferrocene is correctly installed and the
-environment is correctly set.
+**Environment constraints:** rustfmt is installed.
 
 **Description:**
 
-1. The `user` calls `rustc` with the following command line arguments::
+1. The `user` calls `rustfmt` with the following command line arguments::
 
-    --edition 2021
-    --crate-type staticlib
-    <path>
-    # where <path> is the path to the root of the compilation unit, as a
-    # positional argument.
+    <path>...
+    # where <path> is the path to a Rust source file.
 
-2. `rustc` parses the command line arguments.
+2. `rustfmt` generates Rust source files with the same paths. Those files may
+   or may not be modified.
 
-3. `rustc` parses the Rust compilation unit.
+Check formatting of a file
+--------------------------
 
-4. `rustc` analyzes the Rust compilation unit.
+.. id:: RUSTFMT_UC2_CHECK
 
-5. `rustc` generates LLVM IR for the Rust compilation unit.
+**Actor(s):** User, rustfmt.
 
-6. `rustc` invokes `LLVM`, passing the generated LLVM IR along with
-   LLVM-related arguments.
+**Input:** One or multiple Rust source files.
 
-7. `LLVM` generates a C-compatible static library.
+**Output:** A terminal status code.
```

Comment:
better wording is maybe "exit status code"

is it worth mentioning that the diff is also produced (in the case of a failure status code)

Reply:
I am not sure. Because `rustfmt` does not produce the diff. `git` does.

Reply:
When running `--check` the diff is applied by rustfmt, not by git.

Reply:
True